src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -450,26 +453,27 @@ private void MakeFrames(ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
             return result;
         }
 
-        private LambdaFrame GetFrameForScope(BoundNode scope, ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
+        private LambdaFrame GetFrameForScope(Analysis.Scope scope, ArrayBuilder<ClosureDebugInfo> closureDebugInfo)
         {
+            var scopeBoundNode = scope.BoundNode;
             LambdaFrame frame;
-            if (!_frames.TryGetValue(scope, out frame))
+            if (!_frames.TryGetValue(scopeBoundNode, out frame))
             {
-                var syntax = scope.Syntax;
+                var syntax = scopeBoundNode.Syntax;
                 Debug.Assert(syntax != null);
 
                 DebugId methodId = GetTopLevelMethodId();
                 DebugId closureId = GetClosureId(syntax, closureDebugInfo);
 
-                var canBeStruct = !_analysis.ScopesThatCantBeStructs.Contains(scope);
+                var canBeStruct = !_analysis.ScopesThatCantBeStructs.Contains(scopeBoundNode);
 
-                var containingMethod = _analysis.ScopeOwner[scope];
+                var containingMethod = scope.ContainingClosureOpt?.OriginalMethodSymbol ?? _topLevelMethod;
```

Comment:
Could I ask why the design went with using the top-level scope has a null ContainingClosure, instead of an always-empty Scope with appropriate OriginalMethodSymbol? I'm not sure which would introduce more special cases (certainly a special case here, but I'm not sure how many would get introduced if the reverse was implemented)